scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -937,9 +937,17 @@ class TypedPipeInst[T] private[scalding] (@transient inpipe: Pipe,
       // for us. So unwind until you hit the first filter, snapshot,
       // then apply the unwound functions
       case Some((tap, fields, Converter(conv))) =>
-        Execution.from(new Iterable[T] {
-          def iterator = m.openForRead(conf, tap).asScala.map(tup => conv(tup.selectEntry(fields)))
-        })
+        /*
+           * We create a Stream here so we are lazy with construction, but we don't read twice.
+           */
+        val tryStream = scala.util.Try(m.openForRead(conf, tap)
+          .asScala
+          .map(tup => conv(tup.selectEntry(fields)))
+          .toStream)
```

Comment:
Also note: cascading should use the distributed cache for large JobConfs (if not, that's a bug). On the other hand, I guess it will deserialize it once per map-task rather than once per machine (not sure if once per machine will even be doable with Hadoop since I don't know how it starts tasks).